tests/testthat/test-renderer1-knit-print.R

```diff
@@ -1,168 +1,197 @@
+  show_hide <- remDr$Runtime$evaluate(sprintf("div = document.getElementById('%s');div.getElementsByClassName('table.legend tr.label_variable')[0]", as.character(id)),returnByValue = TRUE)$result$value
+  widget <- remDr$Runtime$evaluate(sprintf("div = document.getElementById('%s');div.getElementsByClassName('table.legend tr.label_variable')[1]", as.character(id)),returnByValue = TRUE)$result$value
+    
+  
+  list(a178=a,
+       b934=b,
+       show_hide=show_hide,
+       widget=widget)
+}
+
+plot1top <- get_elements("plot1top")
+plot1bottom <- get_elements("plot1bottom")
+
+# get_circles() returns a list of circles so comparing with list(expected, expected)
+
+test_that("clicking top legend adds/remove points", {
+  expect_equal(get_circles(), list(10, 10))
+  
+  clickID("plot1top_q_label_variable_a178")
+  expect_equal(get_circles(), list(5, 10))
+  
+  clickID("plot1top_q_label_variable_b934")
+  expect_equal(get_circles(), list(0, 10))
+  
+  clickID("plot1top_q_label_variable_b934")
+  expect_equal(get_circles(), list(5, 10))
+  
+  clickID("plot1top_q_label_variable_a178")
+  expect_equal(get_circles(), list(10, 10))
+})
+
+test_that("clicking bottom legend adds/remove points", {
+  expect_equal(get_circles(), list(10, 10))
+  
+  clickID("plot1bottom_q_label_variable_a178")
+  expect_equal(get_circles(), list(10, 5))
+  
+  clickID("plot1bottom_q_label_variable_b934")
+  expect_equal(get_circles(), list(10, 0))
+  
+  clickID("plot1bottom_q_label_variable_b934")
+  expect_equal(get_circles(), list(10, 5))
+  
+  clickID("plot1bottom_q_label_variable_a178")
+  expect_equal(get_circles(), list(10, 10))
+})
+
+clickTop <- function() {
+  remDr$Runtime$evaluate(sprintf("div = document.getElementById('%s');div.getElementsByClassName('show_hide_selector_widgets')[0].dispatchEvent(new CustomEvent('click'));", as.character("plot1top")))
+  remDr$Runtime$evaluate(sprintf("div = document.getElementById('%s'); div.getElementsByClassName('selectize-input')[0].dispatchEvent(new CustomEvent('click'));", as.character("plot1top")))
+
+}
+
+clickBottom <- function() {
+  remDr$Runtime$evaluate(sprintf("div = document.getElementById('%s');div.getElementsByClassName('show_hide_selector_widgets')[0].dispatchEvent(new CustomEvent('click'));", as.character("plot1bottom")))
+  remDr$Runtime$evaluate(sprintf("div = document.getElementById('%s'); div.getElementsByClassName('selectize-input')[0].dispatchEvent(new CustomEvent('click'));", as.character("plot1bottom")))
+  
+}
+
+# Function to send a key event
+sendKey <- function(key, code, keyCode) {
+  remDr$Input$dispatchKeyEvent(type = "keyDown", key = key, code = code, windowsVirtualKeyCode = keyCode, nativeVirtualKeyCode = keyCode)
+  remDr$Input$dispatchKeyEvent(type = "keyUp", key = key, code = code, windowsVirtualKeyCode = keyCode, nativeVirtualKeyCode = keyCode)
+}
+
+sendBackspace <- function() {
+  sendKey("Backspace", "Backspace", 8)
+  
```

Comment:
please avoid empty lines in the middle of functions

Reply:
removed them